test/jit/test_profiler.py

```diff
@@ -65,3 +65,21 @@ def test_fuse(a, b):
         optimized_block = next(g.findNode("prim::If").blocks())
         # broadcasts occurred, currently expect to see aten::_grad_sum_to_size
         self.assertIsNotNone(optimized_block.findNode("aten::_grad_sum_to_size"))
+
+    def test_specialized_types(self):
```

Comment:
Shouldn't existing tests be already testing this? I'd prefer to not adding redundancy unless we're truly testing something unique here.

Reply:
I'm testing specifically that the specialized types exist for the nodes we expect them too (TypeCheck / TensorExpr) and not the other nodes (e.g. prim::If). I think that's a new set of tests. `test_fuse` could probably be renamed